compiler/shared/test/scala/mlscript/compiler/TestIR.scala

```diff
@@ -0,0 +1,58 @@
+package mlscript.compiler
+
+
+import mlscript.utils.shorthands._
+import mlscript.compiler.ir._
+import scala.collection.mutable.StringBuilder
+import mlscript.{DiffTests, ModeType, TypingUnit}
+import mlscript.compiler.ir.{Interpreter, Fresh, FreshInt, Builder}
+
+class IRDiffTestCompiler extends DiffTests {
+  import IRDiffTestCompiler.*
+
+  override def postProcess(mode: ModeType, basePath: List[Str], testName: Str, unit: TypingUnit): List[Str] = 
```

Comment:
Instead of building a huge output all at once, use the same trick as in https://github.com/hkust-taco/mlscript/pull/196/files#diff-66a069f49a1bbe4e47568249d4a6035441cb8750933deb676a83c623bb1b1e0aR57

Reply:
The trick is not yet merged into the upstream. Are you suggesting I repeat the modification in that PR here?

Reply:
Yes, indeed. Just add the change on your branch too. It will be trivial to merge later.